cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/type/persondraw/PersonDrawWidget.kt

```diff
@@ -0,0 +1,97 @@
+package com.cognifide.cogboard.widget.type.persondraw
+
+import com.cognifide.cogboard.CogboardConstants
+import com.cognifide.cogboard.config.service.BoardsConfigService
+import com.cognifide.cogboard.storage.ContentRepository
+import com.cognifide.cogboard.widget.BaseWidget
+import io.vertx.core.Vertx
+import io.vertx.core.json.JsonArray
+import io.vertx.core.json.JsonObject
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+
+class PersonDrawWidget(vertx: Vertx, config: JsonObject, boardService: BoardsConfigService = BoardsConfigService()) :
+        BaseWidget(vertx, config, boardService) {
+
+    init {
+        super.config.put(CogboardConstants.PROP_SCHEDULE_PERIOD, SYNC_INTERVAL)
+        createDynamicChangeSubscriber()
+                .handler { userEventCycle() }
+    }
+
+    override fun updateState() {
+        timeoutCycle()
+    }
+
+    private fun timeoutCycle() {
+        cycle(forceCycle = false)
+    }
+
+    private fun userEventCycle() {
+        cycle(forceCycle = true)
+    }
+
+    private fun cycle(forceCycle: Boolean = false) {
+        val isDaily = config.getBoolean(PROP_IS_DAILY, false)
+        val values = config.getJsonArray(PROP_VALUES, JsonArray()).filterIsInstance<String>()
+        val randomize = config.getBoolean(PROP_RANDOMIZE, false)
+        val interval = config.getLong(PROP_INTERVAL, SELECT_INTERVAL)
```

Comment:
Could those be a fields of this class? do we need to load them each time?

Reply:
Those values could be changed manually in dialog after widget instance in created. If life-cycle of widget guaranties that changing configuration will execute `init` method I will fix this.

Reply:
Fixed